### ISSUE TITLE
build: check-snapshot-anchors INLINE_ANCHOR_RE misses insta non-raw @"…" inline anchor

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#876]
## Summary

`INLINE_ANCHOR_RE` only recognizes the raw-string inline-snapshot form `@r"…"` / `@r#"…"#` / `@r###"…"###`, missing insta's non-raw inline form `@"…"`; a snapshot anchored with `@"…"` is mis-classified as bare, producing a spurious gate failure.

## Location

- `check-snapshot-anchors.py:38` — `INLINE_ANCHOR_RE = re.compile(r"@r#*\"")`
- consumed at `:203` in `count_bare` (`if INLINE_ANCHOR_RE.search(body): continue`)

## Evidence

```python
INLINE_ANCHOR_RE = re.compile(r"@r#*\"")
```

The pattern requires a literal `r` after `@`. insta accepts both `@r"…"` (raw) and `@"…"` (non-raw) as inline snapshot anchors. Confirmed:

```
INLINE_ANCHOR_RE.search('@r"foo"')   -> match
INLINE_ANCHOR_RE.search('@r###"foo"') -> match
INLINE_ANCHOR_RE.search('@"foo"')    -> None   # non-raw inline missed
```

A test such as `insta::assert_json_snapshot!(metric.x, @"123")` is a fully anchored inline snapshot, but `count_bare` would not see the anchor and would count it as bare.

## Expected Behavior

Both `@r…"` and `@"` inline forms should be recognized as anchors, e.g. `re.compile(r'@r?#*"')`.

## Actual Behavior

Only the raw-string `@r…"` form is recognized; a `@"…"` inline snapshot is counted as bare and can trip the regression gate.

## Impact

False positive (spurious CI failure) for a legitimate inline-anchored snapshot. Lower severity than a gate bypass: the project conventionally uses `@r#"…"#` (the AGENTS.md "Validation gates" example shows `@r###"…"###`), so no current metric file uses `@"…"`. Still, the policy's intent is "an inline expected block", which `@"…"` satisfies; the gate should not penalize a valid anchor form.

**Comment by dekobon** [comment]
## General Plan for Fix

### Root Cause

`INLINE_ANCHOR_RE = re.compile(r"@r#*\"")` hardcodes the `r` of the raw-string inline form, so insta's non-raw `@"…"` anchor never matches.

### Implementation Steps

1. Make the `r` optional: `INLINE_ANCHOR_RE = re.compile(r'@r?#*"')`. This matches `@"`, `@r"`, `@r#"`, `@r###"`, etc. (Note: `#*` after an absent `r` cannot legally precede a non-raw `@"`, but allowing it is harmless — insta source never produces `@#"`.)

### Tests to Add/Update

- Unit test: `count_bare`-equivalent over a snapshot anchored with `@"123"` — assert it is NOT counted as bare.
- Keep an existing `@r###"…"###` case to confirm no regression.

### Verification

- [ ] `python3 check-snapshot-anchors.py --show` unchanged on the current tree.
- [ ] New test passes.
- [ ] `make snapshot-anchors` green.

**Comment by dekobon** [comment]
Fixed in commit `dc6ff89e` (branch `worktree-agent-a1f97f28013f5d40d`, not yet pushed).

`INLINE_ANCHOR_RE` changed from `@r#*"` to `@r?#*"`, so the `r` is optional and insta's non-raw `@"…"` inline anchor is recognized rather than mis-counted as bare. Added a `--self-test` case asserting `insta::assert_json_snapshot!(m.x, @"123")` is not counted as bare (plus a `@r###"…"###` regression case); the non-raw case fails against the old regex and passes against the new. `make snapshot-anchors` stays green. Not closing per instruction.
